components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/dao/ParMgtDAOImpl.java

```diff
@@ -41,94 +41,91 @@
 
             try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.
                     ParSQLQueries.STORE_PAR_REQUEST)) {
 
-                prepStmt.setString(1, reqUriUUID);
+                prepStmt.setString(1, requestURIReference);
                 prepStmt.setString(2, clientId);
-                prepStmt.setLong(3, scheduledExpiryTime);
-                prepStmt.setString(4, getJsonParams(parameters));
+                prepStmt.setLong(3, expiresIn);
+                prepStmt.setString(4, getSerializedParams(parameters));
 
                 prepStmt.execute();
             } catch (SQLException e) {
                 IdentityDatabaseUtil.rollbackTransaction(connection);
-                throw new ParCoreException("Error occurred in persisting the successful PAR request with" +
-                        " uuid: " + reqUriUUID);
+                throw new ParCoreException("Error occurred in persisting PAR request.", e);
             }
         } catch (SQLException e) {
-            throw new ParCoreException("Error occurred in persisting the successful PAR request with" +
-                    " uuid: " + reqUriUUID);
+            throw new ParCoreException("Error occurred in persisting PAR request.", e);
         }
     }
 
     @Override
-    public ParRequestDO getParRequest(String reqUriUUID) throws ParCoreException {
+    public ParRequestDO getRequestData(String requestURIReference) throws ParCoreException {
 
         try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
              PreparedStatement prepStmt = connection.prepareStatement(SQLQueries
                      .ParSQLQueries.RETRIEVE_PAR_REQUEST)) {
 
-            prepStmt.setString(1, reqUriUUID);
+            prepStmt.setString(1, requestURIReference);
 
             try (ResultSet resultSet = prepStmt.executeQuery()) {
                 if (resultSet.next()) {
                     if (log.isDebugEnabled()) {
-                        log.debug("Successfully obtained client_id of RequestURI with UUID: " + reqUriUUID);
+                        log.debug("Successfully obtained client_id of RequestURI with UUID: " + requestURIReference);
                     }
-                    String jsonParams = resultSet.getString(ParConstants.COL_LBL_JSON_PARAMS);
+                    String jsonParams = resultSet.getString(ParConstants.COL_LBL_PARAMETERS);
                     long scheduledExpiry = resultSet.getLong(ParConstants.COL_LBL_SCHEDULED_EXPIRY);
                     String clientId = resultSet.getString(ParConstants.COL_LBL_CLIENT_ID);
 
-                    return new ParRequestDO(getParams(jsonParams), scheduledExpiry, clientId);
+                    return new ParRequestDO(getDeserializedParams(jsonParams), scheduledExpiry, clientId);
 
                 } else {
                     throw new ParCoreException("uuid does not exist in the database");
```

Comment:
no need to throw an exception here. Handle the empty response at the service level.